src/elf/mod.rs

```diff
@@ -300,25 +300,20 @@ if_sylvan! {
                 dynsyms = Symtab::parse(bytes, dyn_info.symtab, num_syms, ctx)?;
             }
 
-            // iterate through shdrs again iff we're an ET_REL
-            let shdr_relocs = {
-                let mut relocs = vec![];
-                if header.e_type == header::ET_REL {
-                    for (idx, section) in section_headers.iter().enumerate() {
-                        if section.sh_type == section_header::SHT_REL {
-                            section.check_size(bytes.len())?;
-                            let sh_relocs = RelocSection::parse(bytes, section.sh_offset as usize, section.sh_size as usize, false, ctx)?;
-                            relocs.push((idx, sh_relocs));
-                        }
-                        if section.sh_type == section_header::SHT_RELA {
-                            section.check_size(bytes.len())?;
-                            let sh_relocs = RelocSection::parse(bytes, section.sh_offset as usize, section.sh_size as usize, true, ctx)?;
-                            relocs.push((idx, sh_relocs));
-                        }
-                    }
+            let mut shdr_relocs = vec![];
+            for (idx, section) in section_headers.iter().enumerate() {
+                if section.sh_type == section_header::SHT_REL {
```

Comment:
Random thought, you don’t have to do it, wish we just did:

```
let is_rela = section.sh_type == section_header::SHT_RELA
if is_rela && section.sh_type == section_header::SHT_REL {
  // code and pass is_rela to the Boolean flag in parse
}
```

Reply:
Yes, that seems like a better idea and it's what I have in my work around for section relocations